react-common/components/controls/Button.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import { classList, ControlProps } from "../util";
+
+interface ButtonProps extends ControlProps {
+    onClick: () => void;
+    title: string;
+    label?: string;
+    leftIcon?: string;
+    rightIcon?: string;
+    disabled?: boolean;
+    href?: string;
+    target?: string;
+}
+
+export const Button = (props: ButtonProps) => {
+    const {
+        id,
+        className,
+        ariaLabel,
+        ariaHidden,
+        role,
+        onClick,
+        title,
+        label,
+        leftIcon,
+        rightIcon,
+        disabled,
+        href,
+        target
+    } = props;
+
+    const classes = classList(
+        "common-button",
+        className,
+        disabled && "disabled"
+    );
+
+    let clickHandler = () => {
+        if (onClick) onClick();
+        if (href) window.open(href, target || "_blank", "noopener,noreferrer")
+    }
+
+    return (
+        <button
+            id={id}
+            className={classes}
+            title={title}
```

Comment:
could have the title default to the label value unless otherwise specified? i think this is the behavior we generally want but it might be a little too much behind the scenes, idk

Reply:
^^ I think this would be good myself, we've historically missed a LOT of titles & had a lot of cases where it's just set to aria-label anyways

Reply:
So title is a required property and label is not. I did it that way so that icon buttons would always have a title, even when they don't have a label (e.g. they're just an icon). I can split this into two components if we want (Button and IconButton) and make label required in one and title required in the other.

Reply:
hmm yeah i was trying to think of a way to avoid the duplication of the same string in the title/label all over the place but i do think i prefer to have a single button component.... not very strongly either way though

Reply:
Alright I'm going to stick with it as-is and we'll see if it's to annoying